### PR TITLE
Fix/263 radio group

### LINKED ISSUE #263: [Component] - Radio group
Figma - https://www.figma.com/file/XFbhstkgQFz8ZefAU3w2p4/1.-Quantum-Design-System?type=design&node-id=67-737&mode=design

Storybook - https://nearform.github.io/quantum/?path=/docs/form-radiogroup--docs

1.  There is too little spacing between the radio button and the label. Please ensure there is 8px spacing here.
2. The vertical alignment looks slightly off (label looks 1-2px too low). Please ensure the radio button and the label line up vertically
3. Check the spacing between each row. There should be 12 px

### LINKED ISSUE #293: [Component] - ControlLabel
We need a better handling of the labels. At the moment, Quantum provides a `Label` component and many different form components, like `Switch`, `Radio`, and `Checkbox`. 

If you want to add a label to these fields, you have to do it manually: 
- you have to create a `div` container, adding some Tailwind CSS classes, like `flex`, horizontal and vertical spacing, etc...
- add to this `div` the `Label` and `Switch`/`Radio`/`Checkbox` components

This is what happens now in the `Radio` or `Switch` storybooks. We create "demo" components that add labels to those input elements. 
Let us avoid reinventing the wheel every time and provide a better handling of input labels.

I checked some of the existing UI toolkits:
- NetxUI: set the label as a child of the input component: `<Radio value="buenos-aires">Buenos Aires</Radio>` [[link](https://nextui.org/docs/components/radio-group)]
- Blueprint: use a label prop: ` <Radio label="Salad" value="two" />` [[link](https://blueprintjs.com/docs/#core/components/radio)]
- MUI: use a wrapper component `<FormControlLabel value="male" control={<Radio />} label="Male" />` [[link](https://mui.com/material-ui/react-radio-button/)]

Of the three options, I think the best is the MUI one for 2 reasons:
- Label and input are separated. You can use a `<Radio>` without a label.
- We do not pollute the input properties if we want to customize the label, for example, by adding a "hint" text or a label position.

So, my idea is to:
- keep the keep the `Label` and input components independent
- provide a new component called "ControlLabel": `<ControlLabel label="Male"><Radio value="male" /></ControlLabel>`

- [ ] Create component
- [ ] Create Storybook demo
- [ ] Write Storybook docs
- [ ] Implement dark mode

### PR DESCRIPTION
**Pull request by andreaforni**
Closes #263 
Closes #293

This PR:
- Fixes the radio group spacing problems described in #263 
- Introduces the new `ControlLabel` component (described in #293) and:
  - uses it in the `Radio` and `Swicth` stories
  - adds a storybook story for it

Screenshot 2024-01-10 at 09 13 24

Screenshot 2024-01-10 at 09 52 11

### CODE REVIEW

File: src/components/index.ts
Comment (andreaforni) [inline src/components/index.ts:1]: You see a lot of changes here but I've only added the `ControlLabel` import and sorted the imports

**Comment by nickzafiropulos** [comment]
Looks good. Thanks!

**Comment by andreaforni** [comment]
> Looks good on the radio buttons. Could you add a left/right/Top (position) property for the Control label? @nickzafiropulos what do you think

Good point, Marcelo. I can add the label position, but I'd like to have some examples or to understand if we really need it based on the current Figma design.

**Comment by andreaforni** [comment]
> > Looks good on the radio buttons. Could you add a left/right/Top (position) property for the Control label? @nickzafiropulos what do you think
> 
> I Agree. I don't know how often we would use Top, but it doesn't hurt. Left/Right is good though. I'll add it to the design backlog to update the Figma component to allow variants that show left/right/top labels

Perfect, I'll add the Left/Right position for now.

**Comment by marceloFerreira90** [comment]
ok I think its a good point. I am going to approve this and raise a new ticket based off.of nicks comment

**Review by marceloFerreira90 — APPROVED** [review]

**Pull request merged**